### PR TITLE
Adds the Tider taser

### PR DESCRIPTION
**Pull request by JoshAdamPowell**
## About The Pull Request

Adds an admin-only taser that only works on assistants.
Oh, and if they're hit in the bridge or the Captain's office they'll be gibbed.
Also also, if you try and use it as an assistant, that's a gibbin'
Based off goof's PR: https://github.com/tgstation/tgstation/pull/66873

## Why It's Good For The Game

If the tide is being particularly annoying, gives admins a way to ICly tip the balance in security's favour.

## Changelog

:cl: WineAllWine
admin: admins can give out the tider taser
/:cl:

### CODE REVIEW

**Comment by Shadowflame909** [comment]
The Grey tide..and now the tider-taser..

Nanotransen wages war across station.. Unseen in normal facilities

**Comment by ArcaneMusic** [comment]
This looks like it still works if the player changes jobs from assistant, should it?

**Comment by JoshAdamPowell** [comment]
> This looks like it still works if the player changes jobs from assistant, should it?

Once a tider always a tider

File: code/modules/projectiles/projectile/energy/stun.dm [hunk 1]
```diff
@@ -25,6 +28,42 @@
...
+	. = ..()
+	var/mob/living/carbon/C = target
+	var/area/Location = get_area(C)
+	var/list/gibbable_areas = list(/area/station/command/bridge, /area/station/command/heads_quarters/captain)
+	for(var/area/A as anything in gibbable_areas)
+		if (istype(Location, A))
+			C.gib()
+			return
+	C.Paralyze(10 SECONDS)
+	C.set_timed_status_effect(10 SECONDS, /datum/status_effect/speech/stutter)
```
Comment (SmArtKar) [inline code/modules/projectiles/projectile/energy/stun.dm:53]: use descriptive variable names

**Comment by ArcaneDefence** [comment]
I still think it'd be funnier if it allowed assistants to fire it too, but would taze both the target and the shooter if it hit an assistant

**Comment by iwishforducks** [comment]
this sounds like a really effective item to give to security if there's a shit ton of tiding assistants. but gibbing? could there be a subtype of it that doesn't gib?

File: code/modules/projectiles/guns/energy/stun.dm [hunk 2]
```diff
@@ -6,6 +6,18 @@
...
+/obj/item/gun/energy/taser/tider // admin only. fun.
+	name = "tider taser"
+	desc = "A low-capacity, energy-based stun gun used by security teams to subdue targets at range. This model only works on assistants."
+	ammo_type = list(/obj/item/ammo_casing/energy/electrode/tider)
+
+/obj/item/gun/energy/taser/tider/process_fire(atom/target, mob/living/user)
+	. = ..()
+	if (!user.mind)
+		return
+	if(istype(user.mind.assigned_role, /datum/job/assistant))
```
Comment (Fikou) [inline code/modules/projectiles/guns/energy/stun.dm:18]: use is_assistant_job

File: code/modules/projectiles/projectile/energy/stun.dm [hunk 3]
```diff
@@ -25,6 +28,42 @@
...
+	name = "tider taser"
+
+// We only want to activate on Assistants.
+/obj/projectile/energy/electrode/tider/tase_checks(mob/target)
+	if(!ishuman(target))
+		return FALSE
+	var/mob/living/carbon/human/human_target = target
+	if (!human_target.mind)
+		return FALSE
+	if(istype(human_target.mind.assigned_role, /datum/job/assistant))
```
Comment (Fikou) [inline code/modules/projectiles/projectile/energy/stun.dm:66]: and here

**Comment by JoshAdamPowell** [comment]
> this sounds like a really effective item to give to security if there's a shit ton of tiding assistants. but gibbing? could there be a subtype of it that doesn't gib?

It's admin only, just make another PR if you want a lame tider-taser

**Comment by JohnFulpWillard** [comment]
review comments

File: code/modules/projectiles/projectile/energy/stun.dm [hunk 4]
```diff
@@ -25,6 +28,42 @@
...
+	stutter = 0
+	jitter = 0
+
+
 /obj/projectile/energy/electrode/on_range() //to ensure the bolt sparks when it reaches the end of its range if it didn't hit a target yet
 	do_sparks(1, TRUE, src)
 	..()
+
+/obj/projectile/energy/electrode/tider/on_hit(atom/target)
+	if(!tase_checks(target))
```
Comment (JohnFulpWillard) [inline code/modules/projectiles/projectile/energy/stun.dm:42]: This is the only place it's called, so why is the proc on parent?

File: code/modules/projectiles/projectile/energy/stun.dm [hunk 5]
```diff
@@ -25,6 +28,42 @@
...
+	paralyze = 0
+	stutter = 0
+	jitter = 0
+
+
 /obj/projectile/energy/electrode/on_range() //to ensure the bolt sparks when it reaches the end of its range if it didn't hit a target yet
 	do_sparks(1, TRUE, src)
 	..()
+
+/obj/projectile/energy/electrode/tider/on_hit(atom/target)
```
Comment (JohnFulpWillard) [inline code/modules/projectiles/projectile/energy/stun.dm:41]: Don't separate this with the bullet itself please

**Comment by optimumtact** [comment]
open a new version fo this when you're actually gonna work on it.

**Pull request closed without merging**